cbor2/decoder.py

```diff
@@ -117,7 +125,13 @@ def decode_semantic(decoder, subtype, shareable_index=None):
         shareable_index = decoder._allocate_shareable()
         return decoder.decode(shareable_index)
 
-    value = decoder.decode()
+    # Special handling for sets
```

Comment:
This code looks out of place. Why is it not in `decode_set()`?

Reply:
The values must be hashable before passing to decode_set()

Reply:
I'm not convinced yet but I will take a better look this weekend.